tests/src/python/test_qgsgeometry.py

```diff
@@ -1347,135 +1347,142 @@ def testBoundingBox(self):
         assert line.boundingBox().isNull()
 
     def testAddPart(self):
+        geometries = [QgsGeometry.fromPoint(QgsPoint(0, 0), QgsGeometry.fromPoint(QgsPoint(1, 1)]
+        geometry=QgsGeometry.collectGeometry(geometries)
+        expwkt="MultiPoint ((0 0), (1 1))"
+        wkt=geometry.exportToWkt()
+        assert compareWkt(expwkt, wkt), "Expected:\n%s\nGot:\n%s\n" % (expwkt, wkt)
```

Comment:
Hate to be a pain here, but can you extend these to include tests for linestrings/polygons (ie making sure that MultiLineStrings/MultiPolygons are returned), and also add corner cases such as calling the function with an empty list.

Tests to ensure that multi* geometries are returned when calling the function with a single geometry list would be good too.